### ISSUE TITLE
Empty CMAKE_BUILD_TYPE defaults to BETA

### ISSUE THREAD

**Issue by dvzrv** [csound/csound#1165]
Hi all!
I've just stumbled upon, that by default (when [CMAKE_BUILD_TYPE](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html) is empty) [a beta build is forced](https://github.com/csound/csound/blob/develop/CMakeLists.txt#L159-L166).

It would be great, if this could somehow be changed, as this is a very strange default behavior.
E.g. a `Beta` build type could be introduced specifically using [CMAKE_CONFIGURATION_TYPES](https://cmake.org/cmake/help/latest/variable/CMAKE_CONFIGURATION_TYPES.html#variable:CMAKE_CONFIGURATION_TYPES). That would be more specific and clear and also not force non-default behavior during a 'standard' build, where a user would normally assume to build a standard/ empty target, when not defining `CMAKE_BUILD_TYPE`.

**Comment by vlazzarini** [comment]
I have the impression this is a CMake thing, when build type is not set, it defaults to Debug. BETA is defined for Debug builds.

**Comment by stekyne** [comment]
It seems this is set internally in a few places too: https://github.com/csound/csound/blob/a491dc0094d34e86a6bcd307a4c8b88ed9d87886/Custom.cmake.ex

But also in the bela custom.cmake:
https://github.com/csound/csound/blob/a491dc0094d34e86a6bcd307a4c8b88ed9d87886/Bela/Custom.cmake.bela

https://github.com/csound/csound/blob/711bcf9ccbcf376a57488a50c73c3aa1428fc5cd/CMakeLists.txt#L151. The custom.cmake file is included right before the code you mentioned so I imagine that is setting it.

**Comment by vlazzarini** [comment]
custom.cmake does not count. That is a user-defined script, not set by default.

**Comment by dvzrv** [comment]
> I have the impression this is a CMake thing, when build type is not set, it defaults to Debug. BETA is defined for Debug builds.

AFAIK, by default it defaults to nothing actually. `Debug` is one of the standard `CMAKE_BUILD_TYPES` though!

The custom `BETA` is a bit of an unfortunate default though and @stekyne yes, it is used throughout the codebase!

**Comment by vlazzarini** [comment]
No, the custom.cmake is only used if it exists and it does not exist by default (you see the name of the file is .ex in the repo).

BETA is an important token for us and it is to be defined under Debug. I have made sure this is the case now in git.

**Comment by jpffitch** [comment]
?Is this fixed now after the revised CMakeLists.txt
